cms/models/pagemodel.py

```diff
@@ -827,6 +827,8 @@ def save(self, **kwargs):
 
         if created:
             self.created_by = self.changed_by
+
+        self.clear_cache(menu=True)
```

Comment:
Please create gh tickets for all bugfixes on released versions (3.6 but not 4.0 in this case).
Because is an rc release regression, you don't need a changelog entry.
Regarding the fix, add it to `AddPageForm` instead.